include/secp256k1_frost.h

```diff
@@ -0,0 +1,695 @@
+ *
+ *  After the call, sig will always be initialized. If parsing failed or the
+ *  encoded numbers are out of range, signature verification with it is
+ *  guaranteed to fail for every message and public key.
+ */
+SECP256K1_API int secp256k1_frost_partial_sig_parse(
+    const secp256k1_context* ctx,
+    secp256k1_frost_partial_sig* sig,
+    const unsigned char *in32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Serialize a FROST share
+ *
+ *  Returns: 1 when the share could be serialized, 0 otherwise
+ *  Args:    ctx: a secp256k1 context object
+ *  Out:   out32: pointer to a 32-byte array to store the serialized share
+ *  In:    share: pointer to the share
+ */
+SECP256K1_API int secp256k1_frost_share_serialize(
+    const secp256k1_context* ctx,
+    unsigned char *out32,
+    const secp256k1_frost_share* share
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Parse a FROST share.
+ *
+ *  Returns: 1 when the share could be parsed, 0 otherwise.
+ *  Args:    ctx: a secp256k1 context object
+ *  Out:   share: pointer to a share object
+ *  In:     in32: pointer to the 32-byte share to be parsed
+ */
+SECP256K1_API int secp256k1_frost_share_parse(
+    const secp256k1_context* ctx,
+    secp256k1_frost_share* share,
+    const unsigned char *in32
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3);
+
+/** Creates a key generation share and verifiable secret sharing ("VSS")
+ *  commitments.
+ *
+ *  To generate a key, each participant generates a share for each other
+ *  participant. For example, in the case of 2 particpants, Alice and Bob, they
+ *  each generate 2 shares, distribute 1 share to each other using a secure
+ *  channel, and keep 1 for themselves.
+ *
+ *  Each participant _must_ have a secure channel with each other participant
+ *  with which they can transmit shares to each other.
+ *
+ *  A new session_id32 _must_ be used for each key generation session. For
+ *  example, in the case of 2 participants, Alice and Bob, Alice will generate
+ *  a session_id32 and use it for each of the 2 calls to
+ *  secp256k1_frost_share_gen and Bob will generate a session_id32 and use it
+ *  for each of the 2 calls to secp256k1_frost_share_gen. Both Alice and Bob
+ *  must NOT REUSE there respective session_id32 again for subsequent key
+ *  generation sessions. If Alice and Bob fail to complete this session or
+ *  start a new session to generate a new key, they must NOT REUSE their
+ *  respective session_id32 again, but instead generate a new one. It is
+ *  recommended to always choose session_id32 uniformly at random to avoid
+ *  their reuse.
+ *
+ *  Returns: 0 if the arguments are invalid, 1 otherwise
+ *  Args:            ctx: pointer to a context object initialized for
+ *                        verification
+ *  Out:  vss_commitment: the coefficient commitments. The length of this array
```

Comment:
Note the equality of the length array to the threshold of signers doesn't seem to be enforced in `frost_share_gen` as a early check. In C you would need to iterate over the whole array once, so it might be burdensome and there is no usage of `vss_commitment` array before iterating in the for loop.